### ISSUE TITLE
Cannot use delete/backspace in Input inside <Html> & carot moves to end of field on updates

### ISSUE THREAD

**Issue by tilnea** [pmndrs/drei#244]
- `three` 0.124.0:
- `react-three-fiber` 5.3.11:
- `@react-three/drei` 2.2.18:
- `node` version: 14.15.0
- `npm` (or `yarn`) version: 1.22.5 yarn)

### Problem description:

I use html from drei and inside I have a input field. when adding to the input everything is fine, but I cannot delete characters inside it

### Relevant code:

```
let [name, setName] = useState("A");

return (
<group>
    <Html center={true}>
        <input
            value={name}
            onInput={(event) => setName(event.target.value)}
        />
    </Html>
    <mesh> ... </mesh>
</group>
)
```

**Comment by joshuaellis** [comment]
Hi @tilnea do you have mwe you could share with us please?

**Comment by tilnea** [comment]
I have tried and tried to make a sandbox but I cannot reproduce it outside of my repo.

I have done a sandbox with something that maybe maybe might be related somehow
https://codesandbox.io/s/drei-input-forked-w08dj?file=/src/index.js

If I set my marker after the big T and type letters, the marker will go to the end of the text after one character is added in the input inside the canvas, but not outside of it.

I'm not sure its related with my issue. 
I can aslo tell you that my colleague can delete a character if he clicks backspace 10 times very fast, but for me that does not work either.
I have tried using both onChange and onInput, onChange does not even fire an event, but onInput does.

**Comment by joshuaellis** [comment]
Hi @tilnea,

> I have tried and tried to make a sandbox but I cannot reproduce it outside of my repo.

Can you share the repo? If you're not able too, then I'm unsure how we can help.

> I have done a sandbox with something that maybe maybe might be related somehow

Thanks for this, we'll look into that bit, looking at the `Html` component, I can imagine why this *may* happen.

**Comment by drcmda** [comment]
it looks like the component is re-mounted on every change. im looking into it as well, made a box for quick testing: https://codesandbox.io/s/drei-input-forked-m2drd?file=/src/Html.tsx

**Comment by drcmda** [comment]
as a quick workaround:

```jsx
function Input() {
  let [text, setText] = useState("Text")
  return (
    <>
      <span>Inside of canvas</span>
      <input value={text} onChange={(event) => setText(event.target.value)} />
    </>
  )
}

function Example() {
  return (
    <Html>
      <Input />
    </Html>
  )
}
```

Each Html component opens up a new react-root, so it must have something to do with that since it's working within the root. Still not sure why or how the problem happens.

**Comment by drcmda** [comment]
it's fixed in 2.2.19: https://codesandbox.io/s/drei-input-forked-m2drd

**Comment by tilnea** [comment]
It solved my backspace problem as well! Thank you! :)
